src/main/kotlin/org/jetbrains/research/ml/codetracker/server/TrackerQueryExecutor.kt

```diff
@@ -68,18 +66,22 @@ object TrackerQueryExecutor : QueryExecutor() {
         }
     }
 
-    private fun executeTrackerQuery(
-        request: Request
-    ): String? {
+    private fun executeTrackerQuery(request: Request): String? {
         val response = executeQuery(request)
         if (response.isSuccessful()) {
             return response?.let { it.body?.string() }
         }
         throw IllegalStateException("Unsuccessful server response")
     }
 
-    private fun sendCodeTrackerData(file: File): String? {
-        return executeTrackerQuery(getRequestForSendingDataQuery("data-item", file, CODE_TRACKER_FILE_FIELD))
+    private fun sendCodeTrackerData(files: List<File>): String? {
+        val keys = files.map {
+            val request = getRequestForSendingDataQuery("data-item", it, CODE_TRACKER_FILE_FIELD)
+            val key = executeTrackerQuery(request) ?: return null
+            key
+        }.toSet()
+//      Todo: we should check that all keys received from server are the same, right?
```

Comment:
Sorry, but I did not understand this logic. The server creates a new codetracker item each attempt sending a file. The user has a list of pairs: codetracker and activity-tracker files. In your case, the activity tracker file will be connected only to the first code tracker file... Maybe we should discuss about it a little bit and I have to change the server structure that the server has a possibility to receive a list of the files but if it is the logic will be a little be different, because then we will have a query for creating the code tracker file and another one for updating it.

Reply:
I have fixed it, but I added TODO. @elena-lyulina please, check it. And probably we should test it more

Reply:
Oh, I am not sure that auto-opening files work now...